tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java

```diff
@@ -501,13 +541,160 @@ protected CallbackHandler getCallbackHandler(String username, String realm, Stri
                     } else {
                         ciac.setAuthorized(null);
                     }
+                } else if (callback instanceof PrincipalAuthorizeCallback){
+                    PrincipalAuthorizeCallback pac = (PrincipalAuthorizeCallback) callback;
+                    pac.setAuthorized(true);
                 } else {
                     throw new UnsupportedCallbackException(callback);
                 }
             }
         };
     }
 
+    public static class TestX509Certificate extends X509Certificate {
```

Comment:
I think this test could make use of `SelfSignedX509CertificateAndSigningKey.builder()` to create the cert instead of introducing this new class.

Reply:
True, I have changed it to what you have suggested